### PR TITLE
Development

### PR DESCRIPTION
**Pull request by olgaboiar**
- create Board; 
- create Game Rules;
- create User Interface;
- create Human Player;
- create Easy Computer;

### CODE REVIEW

File: tests/GameRulesTest.cs [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+            
+            Assert.False(_gameRules.Win(_board, _xMarker)); 
+            Assert.False(_gameRules.Win(_board, _oMarker));
+        }
+        
+        [Test]
+        public void ReturnsTrueForTieWhenBoardIsTie()
+        {
+            Assert.True(_gameRules.Tie(_tieBoard));
+            Assert.True(_gameRules.Tie(_tieBoard));
```
Comment (naush) [inline tests/GameRulesTest.cs]: Looks like all the `.Tie` assertions are doubled. I guess this used to take `_xMarker` and `_oMarker`? 😃

File: ttt_csharp/Board.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        {
+            Spots[spot - 1] = marker;
+        }
+
+        public List<object> GetAvailableSpots()
+        {
+            List<object> availableSpots = new List<object>();
+            foreach (var spot in Spots)
+            {
+                if (spot is int)
```
Comment (naush) [inline ttt_csharp/Board.cs]: `spot is int` can be a bit cryptic, I would extract this to a method perhaps called `IsAvailable(spot)`.

File: ttt_csharp/GameRules.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
+namespace ttt_csharp
+{
+    public class GameRules
+    {
+        public bool HorizontalWin(Board board, char marker)
+        {
+            return board.Spots[0].Equals(board.Spots[1]) & board.Spots[1].Equals(board.Spots[2]) & board.Spots[0].Equals(marker)
+                | board.Spots[3].Equals(board.Spots[4]) & board.Spots[4].Equals(board.Spots[5]) & board.Spots[3].Equals(marker)
+                | board.Spots[6].Equals(board.Spots[7]) & board.Spots[7].Equals(board.Spots[8]) & board.Spots[8].Equals(marker);
```
Comment (naush) [inline ttt_csharp/GameRules.cs:9]: Not a big deal, but maybe there are ways to remove some of the duplication here. The only things changing here are the numbers.

File: ttt_csharp/GameRules.cs [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        {
+            return HorizontalWin(board, marker) | VerticalWin(board, marker) | DiagonalWin(board, marker);
+        }
+        
+        public bool Tie(Board board)
+        {
+            return board.GetAvailableSpots().Count == 0;
+        }
+        
+        public bool GameOver(Board board, char marker)
```
Comment (naush) [inline ttt_csharp/GameRules.cs:35]: Does GameOver need to take in a marker?

**Comment by naush** [comment]
I had a few nitpicking comments but in general I thought the code was quite clean. Good work!

File: tests/GameRulesTest.cs (hunk 1 above)
Comment (olgaboiar) [reply tests/GameRulesTest.cs]: yes :) removed duplicate line in https://github.com/olgaboiar/ttt_csharp/pull/1/commits/0df81a04b911b94ff8be42adcdd621eaecb5bf6c

File: ttt_csharp/Board.cs (hunk 2 above)
Comment (olgaboiar) [reply ttt_csharp/Board.cs]: extracted in https://github.com/olgaboiar/ttt_csharp/pull/1/commits/0df81a04b911b94ff8be42adcdd621eaecb5bf6c

File: ttt_csharp/GameRules.cs (hunk 4 above)
Comment (olgaboiar) [reply ttt_csharp/GameRules.cs:35]: removed it in https://github.com/olgaboiar/ttt_csharp/commit/4bbc0695d77673ee7022c5a612f09dc1a5ecf554

**Pull request merged**